### DISCUSSION TITLE
Update `gatsby new` to keep you in the new project directory

### DISCUSSION THREAD

**Discussion by jsumnersmith** [gatsbyjs/gatsby#25931] [Ideas / Feature Requests]
## Summary
When running `gatsby new`, we change into the new site's directory to run the appropriate installations and then [we move back out to the previous directory](https://github.com/gatsbyjs/gatsby/blob/7b1a0f29eb48f5d0b1848baa269c6764a76dc172/packages/gatsby-cli/src/init-starter.ts#L125). Then, in both the CLI output and any of our docs, we tell the user to move back into that directory. I propose that we don't move back into the original directory but leave the users in their new sites' working directory. 

### Basic example
```shell
$parent-directory ~ gatsby new my-great-site
... code runs ...
$my-great-site ~ gatsby develop
```
### Motivation
The current setup likely causes some confusion (our anonymous telemetry data seems to indicate that) and at a gut-level, this feels like an unnecessary step for users. Dropping that directory change should likely improve the DX especially the first time you run `gatsby new`. 

A major assumption to test via discussion here: 
Use Case A: someone runs `gatsby new` they want to immediately begin working on that site
Use Case B: someone runs `gatsby new` and then wants to move on to something else in the parent directory

I'm assuming that Use Case A is way more common than Use Case B. Thoughts?

**Comment by DSchau** [comment]
One thing to note is that this is _definitely_ a breaking change.

I've seen a number of tutorials (notably, our own, e.g. https://www.gatsbyjs.org/docs/quick-start/) that would need augmented to remove the `cd` step. Further still, some users have ingrained this mental model of (as an example) `gatsby new whatever && cd whatever`, which would require some re-training / fixing with this change.

**Comment by ascorbic** [comment]
This should be in v3
